spring-data-mongodb/src/main/java/org/springframework/data/mongodb/observability/MongoObservationCommandListener.java

```diff
@@ -75,11 +81,26 @@ public void commandStarted(CommandStartedEvent event) {
 			log.debug("Found the following observation passed from the mongo context [" + parent + "]");
 		}
 
-		if (parent == null) {
-			return;
+		MongoHandlerContext observationContext = new MongoHandlerContext(connectionString, event, requestContext);
+		observationContext.setRemoteServiceName("mongo");
+
+		Observation observation = MongoObservation.MONGODB_COMMAND_OBSERVATION
+				.observation(this.observationRegistry, () -> observationContext) //
+				.observationConvention(this.observationConvention);
+
+		if (parent != null) {
+			observation.parentObservation(parent);
 		}
 
-		setupObservability(event, requestContext);
+		observation.start();
+
+		requestContext.put(Observation.class, observation);
+		requestContext.put(MongoHandlerContext.class, observationContext);
```

Comment:
You don't need to store the context anymore cause you can retrieve it from an observation